dashboard/app/controllers/openai_evaluate_controller.rb

```diff
@@ -33,11 +33,20 @@ def evaluate
       evaluationCriteria: "Did the student attempt the level?",
     }
 
+    profanity_detected_response = {
+      aiEvaluation: "Profanity detected",
+      evaluationCriteria: "Did the student use profanity?",
+      aiReasoning: "The response contains profanity and could not be evaluated.",
+    }
+
     if level.is_a?(FreeResponse) && student_work.delete(' ').empty?
       no_attempt_response[:aiReasoning] = "The student response was blank."
       # mimic the format of the response from AI
       json_response = {"content" => no_attempt_response.to_json}
       return render(status: :ok, json: json_response)
+    elsif level.is_a?(FreeResponse) && ProfanityFilter.find_potential_profanities(student_work, "en", {})
+      json_response = {"content" => profanity_detected_response.to_json}
+      return render(status: :ok, json: json_response)
```

Comment:
I think we want to be checking student_work for profanity regardless of the level type. If a student's code is profane, we'd want to flag that too eventually.

Reply:
Yeah, thanks for the changes! Let's bump this branch of the conditional after the "no attempt" check for programming levels. That way we don't need to waste profanity filtering on what is essentially starter code that we can trust will be safe.